wsmatrix@martin.zurowietz.de/WorkspacesViewOverride.js

```diff
@@ -7,22 +7,67 @@ var WorkspacesViewOverride = class {
    constructor(settings) {
       this.settings = settings;
       this.overrideOriginalProperties();
+      this._connectSettings();
+      this._handleNumberOfWorkspacesChanged();
    }
 
    destroy() {
+      this._disconnectSettings();
       this.restoreOriginalProperties();
    }
 
+   _connectSettings() {
+      this.settingsHandlerRows = this.settings.connect(
+         'changed::num-rows',
+         this._handleNumberOfWorkspacesChanged.bind(this)
+      );
+
+      this.settingsHandlerColumns = this.settings.connect(
+         'changed::num-columns',
+         this._handleNumberOfWorkspacesChanged.bind(this)
+      );
+   }
+
+   _disconnectSettings() {
+      this.settings.disconnect(this.settingsHandlerRows);
+      this.settings.disconnect(this.settingsHandlerColumns);
+   }
+
+   _handleNumberOfWorkspacesChanged() {
+      this.rows = this.settings.get_int('num-rows');
+      this.columns = this.settings.get_int('num-columns');
+   }
+
    overrideOriginalProperties() {
       WorkspacesView.prototype._overrideProperties = {
          _updateWorkspaceActors: WorkspacesView.prototype._updateWorkspaceActors,
       };
       WorkspacesView.prototype._updateWorkspaceActors = this._updateWorkspaceActors;
+      WorkspacesView.prototype.getRows = this.getRows.bind(this);
+      WorkspacesView.prototype.getColumns = this.getColumns.bind(this);
    }
 
    restoreOriginalProperties() {
       WorkspacesView.prototype._updateWorkspaceActors = WorkspacesView.prototype._overrideProperties._updateWorkspaceActors;
       delete WorkspacesView.prototype._overrideProperties;
+      delete WorkspacesView.prototype.getRows;
+      delete WorkspacesView.prototype.getColumns;
+   }
+
+   setRows(rows) {
```

Comment:
`setRows` and `setColumns` aren't used anywhere, right? If so, they can be removed.